core/sys/linux/file.odin

```diff
@@ -0,0 +1,1697 @@
+	ret := syscall(SYS_mknodat, dirfd, cast(rawptr) name, transmute(c_uint) mode, dev)
+	return Errno(-ret)
+}
+
+/// Creates a directory
+/// Available since Linux 1.0
+/// On ARM64 available since Linux 2.6.16
+mkdir :: proc "contextless" (name: cstring, mode: Mode)->(Errno) {
+	when ARCH == "arm64" {
+		ret := syscall(SYS_mkdirat, AT_FDCWD, cast(rawptr) name, transmute(c_uint) mode)
+		return Errno(-ret)
+	} else {
+		ret := syscall(SYS_mkdir, cast(rawptr) name, transmute(c_uint) mode)
+		return Errno(-ret)
+	}
+}
+
+/// Create a directory relative to specified dirfd
+/// Available since Linux 2.6.16
+mkdirat :: proc "contextless" (dirfd: Fd, name: cstring, mode: Mode)->(Errno) {
+	ret := syscall(SYS_mkdirat, dirfd, cast(rawptr) name, transmute(c_uint) mode)
+	return Errno(-ret)
+}
+
+/// Remove a directory specified by name
+/// Available since Linux 1.0
+/// On ARM64 available since Linux 2.6.16
+rmdir :: proc "contextless" (name: cstring)->(Errno) {
+	when ARCH == "arm64" {
+		ret := syscall(SYS_unlinkat, AT_FDCWD, cast(rawptr) name, transmute(i32) AT_REMOVEDIR)
+		return Errno(-ret)
+	} else {
+		ret := syscall(SYS_rmdir, cast(rawptr) name)
+		return Errno(-ret)
+	}
+}
+
+/// Delete a name, and possible a file it refers to
+/// Available since Linux 1.0
+/// On ARM64 available since Linux 2.6.16
+unlink :: proc "contextless" (name: cstring)->(Errno) {
+	when ARCH == "arm64" {
+		ret := syscall(SYS_unlinkat, AT_FDCWD, cast(rawptr) name, 0)
+		return Errno(-ret)
+	} else {
+		ret := syscall(SYS_unlink, cast(rawptr) name)
+		return Errno(-ret)
+	}
+}
+
+/// Remove a directory entry relative to a directory file descriptor
+/// Available since Linux 2.6.16
+unlinkat :: proc "contextless" (dirfd: Fd, name: cstring, flags: FD_Flags)->(Errno) {
+	ret := syscall(SYS_unlinkat, dirfd, cast(rawptr) name, transmute(i32) flags)
+	return Errno(-ret)
+}
+
+/// Truncate a file to specified length
+/// On 32-bit architectures available since Linux 2.4
+truncate :: proc "contextless" (name: cstring, length: i64)->(Errno) {
+	when LONG_BITS == 32 {
+		ret := syscall(SYS_truncate64, cast(rawptr) name, compat64_arg_pair(length))
+		return Errno(-ret)
+	} else {
+		ret := syscall(SYS_truncate, cast(rawptr) name, compat64_arg_pair(length))
+		return Errno(-ret)
+	}
+}
+
+/// Truncate a file specified by file descriptor to specified length
+/// On 32-bit architectures available since 2.4
+ftruncate :: proc "contextless" (fd: Fd, length: i64)->(Errno) {
+	when LONG_BITS == 32 {
+		ret := syscall(SYS_ftruncate64, fd, compat64_arg_pair(length))
+		return Errno(-ret)
+	} else {
+		ret := syscall(SYS_truncate, fd, compat64_arg_pair(length))
+		return Errno(-ret)
+	}
+}
+
+/// Open file, get the file descriptor
+/// Available since Linux 1.0
+/// On ARM64 available since Linux 2.6.16
+open :: proc "contextless" (name: cstring, flags: OFlags, mode: Mode = {})->(Fd, Errno) {
+	when ARCH == "arm64" {
```

Comment:
Why not `ODIN_ARCH == .arm64`?

Reply:
At the time I decided I was doing that, I was probably thinking that I could put in any architecture name I want, even for those not supported by Odin as of the current time, so I could make this package somewhat future-proof. But then I gave up at the idea

Maybe it's best that I change them back to enums, at least those get checked